Migrate print statements to use new logging infrastructure

## 📋 Issue: Migrate Print Statements to Logging Infrastructure

### Context
Following the implementation of the logging infrastructure in PR #46 (Issue #43), we now need to migrate existing print statements to use the new logging system.

### Current State
- **Total print statements**: 81 occurrences across 5 files
- **Logging infrastructure**: Fully implemented and tested in `src/claude_tasker/logging_config.py`

### 📊 Print Statement Distribution

| File | Count | Priority | Type |
|------|-------|----------|------|
| `src/claude_tasker/prompt_builder.py` | 26 | High | Mixed DEBUG/INFO |
| `src/claude_tasker/cli.py` | 23 | Medium | User interaction + internal |
| `src/claude_tasker/workspace_manager.py` | 19 | High | Mostly DEBUG |
| `src/claude_tasker/workflow_logic.py` | 12 | Medium | Status messages |
| `src/claude_tasker/logging_config.py` | 1 | Low | Error fallback |

## 🎯 Migration Strategy

### Phase 1: High Priority Modules
**Target**: Files with DEBUG prints that affect production performance

#### 1.1 `workspace_manager.py` (19 prints)
```python
# Before
print(f"[DEBUG] Git status check: return code {result.returncode}")

# After
logger.debug("Git status check completed", extra={
    'return_code': result.returncode,
    'command': 'git status --porcelain'
})
```

#### 1.2 `prompt_builder.py` (26 prints)
- Replace DEBUG prints with `logger.debug()`
- Convert status prints to `logger.info()`
- Add structured logging for better analysis

### Phase 2: User Interface Module
**Target**: `cli.py` (23 prints)

Special handling required:
- **Keep**: User-facing output using Rich `console.print()`
- **Migrate**: Internal status messages to `logger.info()`
- **Migrate**: Error messages to `logger.error()`

```python
# User interaction - KEEP as console output
console.print("[bold green]Task completed successfully\![/bold green]")

# Internal status - MIGRATE to logging
logger.info("Processing task", extra={'task_id': task_id})
```

### Phase 3: Workflow Logic
**Target**: `workflow_logic.py` (12 prints)

- Convert workflow status prints to structured logs
- Add operation context for tracing

## 📝 Implementation Checklist

### Module Migration Tasks
- [ ] Add logging import to each module: `from src.claude_tasker.logging_config import get_logger`
- [ ] Initialize module logger: `logger = get_logger(__name__)`
- [ ] Replace print statements according to mapping rules
- [ ] Test module functionality after migration
- [ ] Verify log output in different log levels

### Print Statement Mapping Rules
```python
# Mapping guide
print("[DEBUG] ...")     → logger.debug(...)
print("[INFO] ...")      → logger.info(...)  
print("[WARNING] ...")   → logger.warning(...)
print("[ERROR] ...")     → logger.error(...)
print(f"Status: {msg}")  → logger.info(msg)
print(f"Error: {msg}")   → logger.error(msg)

# User interaction (CLI only)
print("User prompt")     → console.print(...) # Keep for user interaction
```

### Testing Requirements
- [ ] Run existing test suite after each module migration
- [ ] Verify logging output at different levels:
  - `CLAUDE_LOG_LEVEL=DEBUG` - All messages visible
  - `CLAUDE_LOG_LEVEL=INFO` - No debug messages
  - `CLAUDE_LOG_LEVEL=ERROR` - Only errors visible
- [ ] Test structured logging output with `CLAUDE_LOG_JSON=true`
- [ ] Verify no performance regression

## 🎯 Acceptance Criteria
- [ ] All 81 print statements reviewed and migrated/retained as appropriate
- [ ] Each module uses its own logger instance
- [ ] User-facing CLI output remains unchanged
- [ ] All tests passing
- [ ] Logging configuration documented in each module
- [ ] Performance benchmarks show no regression

## 📈 Success Metrics
- ✅ 0 remaining debug print statements
- ✅ All modules using module-specific loggers
- ✅ Structured logging data available for analysis
- ✅ User experience unchanged in CLI
- ✅ Improved debugging capability in production

## 🔧 Technical Details

### Module Template
```python
"""Module description."""

import logging
from src.claude_tasker.logging_config import get_logger

# Initialize module logger
logger = get_logger(__name__)

class SomeClass:
    def some_method(self):
        logger.debug("Method started", extra={'param': value})
        try:
            # ... code ...
            logger.info("Operation successful", extra={'result': result})
        except Exception as e:
            logger.error("Operation failed", exc_info=True, extra={'context': context})
            raise
```

### Environment Setup for Testing
```bash
# Development - verbose logging
export CLAUDE_LOG_LEVEL=DEBUG
export CLAUDE_LOG_COLORS=true

# Production - structured logging
export CLAUDE_LOG_LEVEL=INFO
export CLAUDE_LOG_JSON=true
export CLAUDE_LOG_FILE=/var/log/claude_tasker.log
export CLAUDE_LOG_SANITIZE=true
```

## 🚀 Benefits After Migration
1. **Performance**: No debug output in production
2. **Debugging**: Configurable verbosity
3. **Analysis**: Structured logs for parsing
4. **Security**: Sensitive data redaction
5. **Monitoring**: Integration-ready logs

## 📚 References
- Logging infrastructure PR: #46
- Logging documentation: `docs/logging-integration.md`
- Test suite: `tests/test_logging_config.py`

## 🏷️ Labels
`enhancement`, `logging`, `migration`, `python`

## 📝 Notes
- This completes the second phase of Issue #43
- Should be done module-by-module to ensure stability
- User-facing output in CLI must remain unchanged
- Consider adding performance benchmarks before/after

--- a/src/claude_tasker/cli.py
+++ b/src/claude_tasker/cli.py
@@ -8,6 +8,9 @@ from typing import List, Optional
 from pathlib import Path
 
 from .workflow_logic import WorkflowLogic, WorkflowResult
+from src.claude_tasker.logging_config import get_logger
+
+logger = get_logger(__name__)
 
 
 def parse_issue_range(issue_arg: str) -> tuple[Optional[int], Optional[int]]:
@@ -213,6 +216,9 @@ def print_results_summary(results: List[WorkflowResult]) -> None:
     # Show failed results
     for result in results:
         if not result.success:
+            logger.error(f"Issue processing failed: {result.message}")
+            if result.error_details:
+                logger.error(f"Error details: {result.error_details}")
             print(f"❌ {result.message}")
             if result.error_details:
                 print(f"   Details: {result.error_details}")
@@ -226,11 +232,13 @@ def main() -> int:
     # Validate arguments
     validation_error = validate_arguments(args)
     if validation_error:
+        logger.error(f"Validation error: {validation_error}")
         print(f"Error: {validation_error}", file=sys.stderr)
         return 1
     
     # Check for CLAUDE.md (required for project context)
     if not Path("CLAUDE.md").exists():
+        logger.error("CLAUDE.md not found. Must be run from project root.")
         print("Error: CLAUDE.md not found. Must be run from project root.", file=sys.stderr)
         return 1
     
@@ -319,6 +327,9 @@ def main() -> int:
                 if result.branch_name:
                     print(f"   Branch: {result.branch_name}")
             else:
+                logger.error(f"Bug analysis failed: {result.message}")
+                if result.error_details:
+                    logger.error(f"Error details: {result.error_details}")
                 print(f"❌ {result.message}")
                 if result.error_details:
                     print(f"   Details: {result.error_details}")
@@ -334,10 +345,12 @@ def main() -> int:
             return 1
     
     except KeyboardInterrupt:
+        logger.warning("Operation interrupted by user")
         print("\n⚠️  Operation interrupted by user", file=sys.stderr)
         return 130  # Standard exit code for SIGINT
     
     except Exception as e:
+        logger.error(f"Unexpected error: {e}", exc_info=True)
         print(f"❌ Unexpected error: {e}", file=sys.stderr)
         return 1
 

--- a/src/claude_tasker/prompt_builder.py
+++ b/src/claude_tasker/prompt_builder.py
@@ -6,6 +6,9 @@ import tempfile
 from typing import Dict, Optional, Any
 from pathlib import Path
 from .github_client import IssueData, PRData
+from src.claude_tasker.logging_config import get_logger
+
+logger = get_logger(__name__)
 
 
 class PromptBuilder:
@@ -168,27 +171,27 @@ Keep the response focused and practical. Format as markdown."""
             max_tokens: Maximum tokens for response
             execute_mode: If True, actually execute the prompt with Claude (not just print)
         """
-        print(f"[DEBUG] _execute_llm_tool called with tool={tool_name}, execute_mode={execute_mode}")
+        logger.debug(f"_execute_llm_tool called with tool={tool_name}, execute_mode={execute_mode}")
         try:
             # Build command based on tool
             if tool_name == 'llm':
                 # LLM tool uses stdin for prompts
-                print("[DEBUG] Using llm tool")
+                logger.debug("Using llm tool")
                 cmd = [
                     'llm', 'prompt', '-'
                 ]
             elif tool_name == 'claude':
                 if execute_mode:
                     # Actually execute with Claude to make code changes
-                    print("[DEBUG] Executing Claude in implementation mode")
-                    print(f"[DEBUG] Prompt preview (first 200 chars): {prompt[:200]}...")
+                    logger.debug("Executing Claude in implementation mode")
+                    logger.debug(f"Prompt preview (first 200 chars): {prompt[:200]}...")
                     # Use headless mode (-p) with permission bypass for autonomous execution
                     cmd = [
                         'claude', '-p', '--permission-mode', 'bypassPermissions'
                     ]
                 else:
                     # Just generate/print prompt
-                    print("[DEBUG] Running Claude in prompt generation mode")
+                    logger.debug("Running Claude in prompt generation mode")
                     cmd = [
                         'claude', '--print',
                         '--output-format', 'json',
@@ -200,18 +203,18 @@ Keep the response focused and practical. Format as markdown."""
             # Run with timeout to prevent hanging
             # Pass prompt via stdin for both llm and claude execute mode
             if tool_name == 'llm' or (tool_name == 'claude' and execute_mode):
-                print(f"[DEBUG] Running command: {' '.join(cmd)}")
-                print(f"[DEBUG] Passing prompt via stdin ({len(prompt)} chars)")
-                timeout_val = 180 if execute_mode else 120
+                logger.debug(f"Running command: {' '.join(cmd)}")
+                logger.debug(f"Passing prompt via stdin ({len(prompt)} chars)")
+                timeout_val = 1200 if execute_mode else 120  # 20 minutes for execution, 2 minutes for generation
                 result = subprocess.run(cmd, input=prompt, capture_output=True, text=True, check=False, timeout=timeout_val)
             else:
                 result = subprocess.run(cmd, capture_output=True, text=True, check=False, timeout=120)
             
             if result.returncode == 0:
                 if execute_mode:
-                    print(f"[DEBUG] Claude execution completed successfully")
-                    print(f"[DEBUG] Output length: {len(result.stdout)} chars")
-                    print(f"[DEBUG] Output preview: {result.stdout[:500]}...")
+                    logger.debug(f"Claude execution completed successfully")
+                    logger.debug(f"Output length: {len(result.stdout)} chars")
+                    logger.debug(f"Output preview: {result.stdout[:500]}...")
                 try:
                     return json.loads(result.stdout)
                 except json.JSONDecodeError:
@@ -221,15 +224,15 @@ Keep the response focused and practical. Format as markdown."""
                         'optimized_prompt': result.stdout.strip()
                     }
             else:
-                print(f"[DEBUG] Command failed with return code {result.returncode}")
-                print(f"[DEBUG] stderr: {result.stderr[:500]}")
+                logger.error(f"Command failed with return code {result.returncode}")
+                logger.error(f"stderr: {result.stderr[:500]}")
                 return None
                 
         except subprocess.TimeoutExpired:
-            print("[DEBUG] Command timed out")
+            logger.error("Command timed out")
             return None
         except (FileNotFoundError, json.JSONDecodeError, Exception) as e:
-            print(f"[DEBUG] Error: {e}")
+            logger.error(f"Error: {e}")
             return None
     
     def build_with_llm(self, prompt: str, max_tokens: int = 4000) -> Optional[Dict[str, Any]]:
@@ -344,11 +347,17 @@ Return ONLY the optimized prompt text - no additional commentary or wrapper text
             
             # Stage 3: Execute optimized prompt (if not prompt-only mode)
             if not prompt_only:
-                print("[DEBUG] Stage 3: Executing optimized prompt with Claude")
-                print(f"[DEBUG] Prompt length: {len(optimized_prompt)} characters")
+                logger.debug("Stage 3: Executing optimized prompt with Claude")
+                logger.debug(f"Prompt length: {len(optimized_prompt)} characters")
                 execution_result = self.build_with_claude(optimized_prompt, execute_mode=True)
                 results['execution_result'] = execution_result
-                print(f"[DEBUG] Execution result: {execution_result is not None}")
+                logger.debug(f"Execution result: {execution_result is not None}")
+                
+                # Check if execution actually succeeded
+                if execution_result is None:
+                    results['error'] = "Claude execution failed or timed out - consider increasing timeout or checking Claude availability"
+                    results['success'] = False
+                    return results
             
             results['success'] = True
             return results
@@ -363,7 +372,7 @@ Return ONLY the optimized prompt text - no additional commentary or wrapper text
         This method runs Claude in headless mode and captures the full output
         for PR review comments.
         """
-        print("[DEBUG] Executing Claude for PR review")
+        logger.debug("Executing Claude for PR review")
         try:
             # Write prompt to temporary file to avoid shell escaping issues
             with tempfile.NamedTemporaryFile(mode='w', suffix='.txt', delete=False) as f:
@@ -374,8 +383,8 @@ Return ONLY the optimized prompt text - no additional commentary or wrapper text
                 # Run Claude in headless mode to generate the review
                 cmd = ['claude', '-p', prompt_file, '--permission-mode', 'bypassPermissions']
                 
-                print(f"[DEBUG] Running command: {' '.join(cmd)}")
-                print(f"[DEBUG] Prompt length: {len(prompt)} chars")
+                logger.debug(f"Running command: {' '.join(cmd)}")
+                logger.debug(f"Prompt length: {len(prompt)} chars")
                 
                 # Execute with longer timeout for review generation
                 result = subprocess.run(
@@ -383,13 +392,13 @@ Return ONLY the optimized prompt text - no additional commentary or wrapper text
                     capture_output=True,
                     text=True,
                     check=False,
-                    timeout=300  # 5 minute timeout for reviews
+                    timeout=1200  # 20 minute timeout for reviews
                 )
                 
                 if result.returncode == 0:
                     output = result.stdout.strip()
-                    print(f"[DEBUG] Claude review completed successfully")
-                    print(f"[DEBUG] Output length: {len(output)} chars")
+                    logger.debug(f"Claude review completed successfully")
+                    logger.debug(f"Output length: {len(output)} chars")
                     
                     # For reviews, we want the full output as the response
                     return {
@@ -397,16 +406,16 @@ Return ONLY the optimized prompt text - no additional commentary or wrapper text
                         'success': True
                     }
                 else:
-                    print(f"[DEBUG] Claude review failed with return code {result.returncode}")
-                    print(f"[DEBUG] stderr: {result.stderr[:500]}")
+                    logger.error(f"Claude review failed with return code {result.returncode}")
+                    logger.error(f"stderr: {result.stderr[:500]}")
                     return None
             finally:
                 # Clean up temp file
                 Path(prompt_file).unlink(missing_ok=True)
                 
         except subprocess.TimeoutExpired:
-            print("[DEBUG] Claude review command timed out")
+            logger.error("Claude review command timed out")
             return None
         except Exception as e:
-            print(f"[DEBUG] Error executing Claude review: {e}")
+            logger.error(f"Error executing Claude review: {e}")
             return None

--- a/src/claude_tasker/workflow_logic.py
+++ b/src/claude_tasker/workflow_logic.py
@@ -11,6 +11,9 @@ from .github_client import GitHubClient
 from .workspace_manager import WorkspaceManager
 from .prompt_builder import PromptBuilder
 from .pr_body_generator import PRBodyGenerator
+from src.claude_tasker.logging_config import get_logger
+
+logger = get_logger(__name__)
 
 
 @dataclass
@@ -173,9 +176,9 @@ class WorkflowLogic:
                 )
             
             # Check if there are changes to commit
-            print("[DEBUG] Checking for git changes after Claude execution")
+            logger.debug("Checking for git changes after Claude execution")
             has_changes = self.workspace_manager.has_changes_to_commit()
-            print(f"[DEBUG] Git has changes: {has_changes}")
+            logger.debug(f"Git has changes: {has_changes}")
             
             if has_changes:
                 # Commit changes
@@ -283,6 +286,7 @@ The issue has been reviewed and no further action is needed at this time.
         results = []
         
         for issue_number in range(start_issue, end_issue + 1):
+            logger.info(f"Processing issue #{issue_number}...")
             print(f"\n🔄 Processing issue #{issue_number}...")
             
             result = self.process_single_issue(
@@ -294,6 +298,7 @@ The issue has been reviewed and no further action is needed at this time.
             
             # Apply timeout between issues (except for last one)
             if issue_number < end_issue and self.timeout_between_tasks > 0:
+                logger.debug(f"Waiting {self.timeout_between_tasks} seconds between tasks")
                 print(f"⏳ Waiting {self.timeout_between_tasks} seconds...")
                 time.sleep(self.timeout_between_tasks)
         
@@ -333,10 +338,11 @@ The issue has been reviewed and no further action is needed at this time.
             )
             
             # Execute review (using claude directly for reviews with review_mode=True)
-            print(f"[DEBUG] Generating review for PR #{pr_number}...")
+            logger.debug(f"Generating review for PR #{pr_number}...")
             review_result = self.prompt_builder.build_with_claude(review_prompt, review_mode=True)
             
             if not review_result:
+                logger.error(f"No review result returned for PR #{pr_number}")
                 print(f"[ERROR] No review result returned for PR #{pr_number}")
                 return WorkflowResult(
                     success=False,
@@ -344,6 +350,8 @@ The issue has been reviewed and no further action is needed at this time.
                 )
             
             if not review_result.get('response'):
+                logger.error(f"Review result missing 'response' field for PR #{pr_number}")
+                logger.debug(f"Review result keys: {review_result.keys()}")
                 print(f"[ERROR] Review result missing 'response' field for PR #{pr_number}")
                 print(f"[DEBUG] Review result keys: {review_result.keys()}")
                 return WorkflowResult(
@@ -355,8 +363,8 @@ The issue has been reviewed and no further action is needed at this time.
             if not prompt_only:
                 # Get the actual review content
                 review_content = review_result.get('response', 'Review completed')
-                print(f"[DEBUG] Review content length: {len(review_content)} chars")
-                print(f"[DEBUG] Review content preview: {review_content[:200]}...")
+                logger.debug(f"Review content length: {len(review_content)} chars")
+                logger.debug(f"Review content preview: {review_content[:200]}...")
                 
                 # Only add wrapper if there's actual content
                 if review_content and review_content != 'Review completed':
@@ -375,7 +383,7 @@ Unable to generate detailed review. Please review the PR manually.
 ---
 🤖 Generated via automated review with [Claude Code](https://claude.ai/code)"""
                 
-                print(f"[DEBUG] Posting review comment to PR #{pr_number}...")
+                logger.debug(f"Posting review comment to PR #{pr_number}...")
                 if self.github_client.comment_on_pr(pr_number, review_comment):
                     return WorkflowResult(
                         success=True,

--- a/src/claude_tasker/workspace_manager.py
+++ b/src/claude_tasker/workspace_manager.py
@@ -5,6 +5,9 @@ import os
 import time
 from typing import Tuple, Optional, List
 from pathlib import Path
+from src.claude_tasker.logging_config import get_logger
+
+logger = get_logger(__name__)
 
 
 class WorkspaceManager:
@@ -130,9 +133,11 @@ class WorkspaceManager:
         # Stash all changes including untracked files
         result = self._run_git_command(['stash', 'push', '-u', '-m', stash_message])
         if result.returncode != 0:
+            logger.error(f"Failed to stash changes: {result.stderr}")
             print(f"Failed to stash changes: {result.stderr}")
             return False
         
+        logger.info(f"Changes stashed successfully: {stash_message}")
         print(f"Changes stashed successfully: {stash_message}")
         print("You can restore them later with: git stash pop")
         return True
@@ -207,52 +212,52 @@ class WorkspaceManager:
     
     def has_changes_to_commit(self) -> bool:
         """Check if there are staged or unstaged changes."""
-        print("[DEBUG] Checking for git changes...")
+        logger.debug("Checking for git changes...")
         
         # Use git status --porcelain for a more reliable check
         result = self._run_git_command(['status', '--porcelain'])
-        print(f"[DEBUG] Git status --porcelain check: return code {result.returncode}")
+        logger.debug(f"Git status --porcelain check: return code {result.returncode}")
         
         if result.returncode != 0:
-            print(f"[DEBUG] Git status command failed: {result.stderr}")
+            logger.debug(f"Git status command failed: {result.stderr}")
             # Fall back to checking individual types of changes
             
             # Check for unstaged changes
             result = self._run_git_command(['diff', '--quiet', 'HEAD'])
-            print(f"[DEBUG] Unstaged changes check (git diff --quiet HEAD): return code {result.returncode}")
+            logger.debug(f"Unstaged changes check (git diff --quiet HEAD): return code {result.returncode}")
             if result.returncode != 0:
                 # Double-check with actual diff output
                 diff_result = self._run_git_command(['diff', 'HEAD'])
                 if diff_result.stdout.strip():
-                    print("[DEBUG] Found unstaged changes")
+                    logger.debug("Found unstaged changes")
                     return True
             
             # Check for staged changes
             result = self._run_git_command(['diff', '--cached', '--quiet'])
-            print(f"[DEBUG] Staged changes check (git diff --cached --quiet): return code {result.returncode}")
+            logger.debug(f"Staged changes check (git diff --cached --quiet): return code {result.returncode}")
             if result.returncode != 0:
                 # Double-check with actual diff output
                 diff_result = self._run_git_command(['diff', '--cached'])
                 if diff_result.stdout.strip():
-                    print("[DEBUG] Found staged changes")
+                    logger.debug("Found staged changes")
                     return True
             
             # Check for untracked files
             result = self._run_git_command(['ls-files', '--others', '--exclude-standard'])
             untracked = result.stdout.strip() if result.returncode == 0 else ""
-            print(f"[DEBUG] Untracked files check: {len(untracked)} chars of output")
+            logger.debug(f"Untracked files check: {len(untracked)} chars of output")
             if untracked:
-                print(f"[DEBUG] Found untracked files: {untracked[:200]}...")
+                logger.debug(f"Found untracked files: {untracked[:200]}...")
                 return True
         else:
             # git status --porcelain succeeded, check its output
             output = result.stdout.strip()
-            print(f"[DEBUG] Git status output length: {len(output)} chars")
+            logger.debug(f"Git status output length: {len(output)} chars")
             if output:
-                print(f"[DEBUG] Git status shows changes:\n{output[:500]}...")
+                logger.debug(f"Git status shows changes:\n{output[:500]}...")
                 return True
             else:
-                print("[DEBUG] Git status shows no changes")
+                logger.debug("Git status shows no changes")
         
         return False
     

--- a/tests/test_critical_edge_cases.py
+++ b/tests/test_critical_edge_cases.py
@@ -142,7 +142,7 @@ class TestCriticalEdgeCases:
             # Simulate timeout
             mock_run.side_effect = subprocess.TimeoutExpired(
                 cmd=['claude', '-p', '--permission-mode', 'bypassPermissions'],
-                timeout=180
+                timeout=1200  # Updated to 20 minutes
             )
             
             # Should handle timeout gracefully
@@ -155,10 +155,10 @@ class TestCriticalEdgeCases:
             # Should return None (failure) not crash
             assert result is None
             
-            # Verify timeout was set appropriately
+            # Verify timeout was set appropriately (updated to 20 minutes)
             mock_run.assert_called_once()
             call_kwargs = mock_run.call_args[1]
-            assert call_kwargs['timeout'] == 180
+            assert call_kwargs['timeout'] == 1200  # 20 minutes
 
     def test_large_prompt_handling(self):
         """Test behavior with very large prompts.

